File: src/main/java/net/alephdev/calendar/models/Team.java
package net.alephdev.calendar.models;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Data;

@Data
@Entity
@Table(name = "Team")
public class Team {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Integer id;

  @Column(nullable = false)
  private String name;

  private String color;

  private String description;

  @Column(name = "is_active", nullable = false)
  private Boolean isActive = true;
}

File: angular/src/app/models/team.ts
/**
 * Модель команды разработки.
 *
 * @property id - Уникальный идентификатор команды
 * @property name - Название команды
 * @property color - Цвет команды (для отображения в UI)
 * @property description - Описание команды
 * @property isActive - Флаг активности команды
 */
export interface Team {
  id: number;
  name: string;
  color: string | null;
  description: string | null;
  isActive: boolean;
}
